Registration/RegisterWithAzure.psm1

```diff
@@ -237,67 +237,175 @@ function Set-AzsRegistration{
         [ValidateNotNull()]
         [string] $MsAssetTag
     )
+
     $ErrorActionPreference = [System.Management.Automation.ActionPreference]::Stop
     $VerbosePreference = [System.Management.Automation.ActionPreference]::Continue
-
     New-RegistrationLogFile -RegistrationFunction $PSCmdlet.MyInvocation.MyCommand.Name
-
     Log-Output "*********************** Begin log: $($PSCmdlet.MyInvocation.MyCommand.Name) ***********************`r`n"
+    
+    $stampOutput = Invoke-RegistrationStep1AgainstPEP -PrivilegedEndpointCredential $PrivilegedEndpointCredential `
+                                                      -PrivilegedEndpoint $PrivilegedEndpoint `
```

Comment:
perhaps we can just ask for the PEP session object, rather than endpoint and creds? Similar to how we ask for the Azure Context, and let the pass that in